src/org/traccar/notificators/NotificatorFirebase.java

```diff
@@ -46,29 +49,38 @@
         private String[] tokens;
         @JsonProperty("notification")
         private Notification notification;
+        @JsonProperty("data")
+        private Map<String, String> data;
     }
 
     public NotificatorFirebase() {
         key = Context.getConfig().getString("notificator.firebase.key");
+        isFormatDataOnly = Context.getConfig().getBoolean("notificator.firebase.dataOnly");
     }
 
     @Override
     public void sendSync(long userId, Event event, Position position) {
         final User user = Context.getPermissionsManager().getUser(userId);
         if (user.getAttributes().containsKey("notificationTokens")) {
 
-            Notification notification = new Notification();
-            notification.body = NotificationFormatter.formatShortMessage(userId, event, position).trim();
-
             Message message = new Message();
             message.tokens = user.getString("notificationTokens").split("[, ]");
-            message.notification = notification;
-
-            Context.getClient().target(URL).request()
+            message.data = NotificationFormatter.buildData(userId, event, position);
+            String body = NotificationFormatter.formatShortMessage(userId, event, position).trim();
+            if (isFormatDataOnly) {
+                message.data.put("body", body);
+            } else {
+                Notification notification = new Notification();
+                notification.body = body;
+                message.notification = notification;
+            }
+
+            ClientBuilder.newClient().target(URL).request()
```

Comment:
Why new client?

Reply:
the problem is that for some reason the json doesn't arrive as it is sent. For example the data I send starts like:
`data: {eventJson={"id":2136,"attributes":{"audio":"20190222102427.amr"}`
as you see eventJson is correct jsonString

but when I receive it it becomes:
`data payload: {eventJson={&#34;id&#34;:2136,&#34;attributes&#34;:{&#34;audio&#34;:&#34;20190222102427.amr&#34;}`

Reply:
this is when it's being sent with the default client